### ISSUE TITLE
Uncaught ValueError in setcookie() after logout

### ISSUE THREAD

**Issue by JavonX** [craigrodway/LoginPersist#12]
I installed the news module version manually. The warnings and deprecated messages after login are now gone.  
However, when I logout, I get these messages:

Deprecated: strlen(): Passing null to parameter #1 ($string) of type string is deprecated in C:\xampp\htdocs\wire\core\WireInput.php on line 371

Deprecated: setcookie(): Passing null to parameter #1 ($name) of type string is deprecated in C:\xampp\htdocs\wire\core\WireInputDataCookie.php on line 414

and this error:
Arrgh… Fatal Error: Uncaught ValueError: setcookie(): Argument #1 ($name) cannot be empty in C:\xampp\htdocs\wire\core\WireInputDataCookie.php:414

**Comment by craigrodway** [comment]
Do you have a cookie name set in this module's Settings > Cookie > Cookie name field?

**Comment by JavonX** [comment]
Yes, the name is "persist".

**Comment by craigrodway** [comment]
Do you still get the error if you remove this module's hook for `logoutSuccess`? Comment out this line to test:

`$this->session->addHookAfter('logoutSuccess', $this, 'hookLoggedOut');`

Do you still get the error if you click Save on the module settings page (even if you don't make any changes)?

The errors you are seeing are complaining that 'null' is being passed as a cookie name - which this module shouldn't be doing.

**Comment by JavonX** [comment]
I don't get an error when I click "Submit" on the modules page without making any changes.

I commented out your suggestet line, refreshed the module cache and logged out => No error or deprecated messages.
Then I uncommented the line again, refreshed the module cache and logged out => No error or deprecated messages.

I assume the issue was fixed when I clicked "Submit" in the module page. The first time I clicked it, it listed a long list of fields that were updated. The second time it only sayed: ProcessModule: Saved Module - LoginPersist (uninstall).

I cannot reproduce the error anymore.
